core/shared/src/main/scala/matryoshka/package.scala

```diff
@@ -104,6 +104,18 @@ package object matryoshka {
     */
   type AlgebraicGTransform[W[_], T, F[_], G[_]]        = F[W[T]] => G[T]
 
+  /** Transform a structure `F` contained in `W`, to a structure `G`,
+    * in bottom-up fashion.
+    * @group algebras
+    */
+  type ElgotAlgebraicTransformM[W[_], M[_], T, F[_], G[_]] = W[F[T]] => M[G[T]]
```

Comment:
I would name these `AlgebraicElgotTransform`[`M`], to match `CoalgebraicElgotTransform`, `AlgebraicGTransform`, etc.